include/mbgl/actor/actor.hpp

```diff
@@ -8,9 +8,108 @@
 #include <memory>
 #include <future>
 #include <type_traits>
+#include <cassert>
 
 namespace mbgl {
 
+template <class Object>
+class EstablishedActor;
+
+template <class Object>
+class Actor;
+
+template <class Object>
+class AspiringActor {
```

Comment:
Put each class in its own header, and add a documentation comment explaining its role.

Reply:
👍  plus some unit tests, now that the design is looking okay